### PR TITLE
doc: add a page with ScyllaDB limits

### PR DESCRIPTION
**Pull request by annastuchlik**
This PR adds a page listing the ScyllDB limits we know today.
The page can and should be extended when other limits are confirmed.

Refs https://github.com/scylladb/scylladb/issues/18897

### CODE REVIEW

File: docs/reference/scylladb-limits.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     - Low tens
+   * - Partition size
+     - Gigabytes
+   * - Rows per partition
+     - No limit
+   * - Row size
+     - Latency-related soft limit: 
+     
+       Hundreds of kilobytes (good latency) or megabytes (mediocre latency)
+   * - CDC
```
Comment (denesb) [inline docs/reference/scylladb-limits.rst:53]: Not clear what this is, is this the number of CDC-enabled tables?

Comment (tzach) [reply docs/reference/scylladb-limits.rst:53]: I sugget adding MV ans SI number per tables as well

File: docs/reference/_common/reference-toc.rst [hunk 2]
```diff
@@ -7,5 +7,6 @@
    GCP Images </reference/gcp-images>
    Configuration Parameters </reference/configuration-parameters>
    Glossary </reference/glossary>
+   ScyllaDB Limits </reference/scylladb-limits>
```
Comment (avikivity) [inline docs/reference/_common/reference-toc.rst:10]: Do we beed ScyllaDB here (and in the URL)? It's all about ScyllaDB.

File: docs/reference/limits.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+   * - Keyspace name length
+     - 48 characters
+   * - Query parameters in a query
+     - 65535 (2^16-1)
+   * - Statements in a batch
+     - 65535 (2^16-1)
+   * - Fields in a tuple
+     - 32768 (2^15) (just a few fields, such as 2-10, are recommended)
+   * - Collection (List)
+     - ~2 billion (231)
```
Comment (denesb) [inline docs/reference/limits.rst:68]: What is this `231`? Should it be `2^31`?
BTW where is this number coming from, 2 billion items in a collection will cause all sort of issues. It won't crash ScyllaDB but tail latencies will probably be terrible.
If these limits are the most ScyllaDB can shoulder without causing unavailability, then this is fine.

Comment (nyh) [reply docs/reference/limits.rst]: By the way, ideally, we shouldn't be blindly documenting hearsay and copying stuff from Cassandra's documentation or grep'ing source code - we should write tests that confirm our beliefs about these limits. In https://github.com/scylladb/scylladb/pull/16779 I wrote tests for the key length issues and learned the differences between what we thought we had, and what we actually have. In https://github.com/scylladb/scylladb/issues/7745 I discovered that although we thought that the length of individual set items was no longer limited to 64KB, sizes above this size don't actually work in unprepared statements (they work in prepared statements). Who knows how many other problems we have like this...

Comment (tzach) [reply docs/reference/limits.rst]: I vote for A. Lets move forward with the data we do know.

Comment (denesb) [reply docs/reference/limits.rst]: Ok, I will queue.

**Comment by denesb** [comment]
@annastuchlik you just need to add the relevant labels and the bot will do the backport.

**Comment by denesb** [comment]
> @denesb Adding the backport label didn't help. Should I manually open a backport PR?

The PR was missing the "backported-to-master" label (which should have been added by the bot). I backported this manually as https://github.com/scylladb/scylladb/commit/3948167fcaced58d21a3aaca21c0ec2cd7741953.

**Pull request closed without merging**